doc/whats_new/upcoming_changes/many-modules/32212.fix.rst

```diff
@@ -0,0 +1,5 @@
+- Raise ValueError when `sample_weight` contains only zero values to prevent
+  ineffective sampling during fitting. This change applies to all estimators that
```

Comment:
Not sure if "ineffective sampling" best describes the scenario. Also does not apply to metrics.

Maybe meaningless input data?